packages/early-boot-config/early-boot-config.spec

```diff
@@ -0,0 +1,152 @@
+Provides: %{_cross_os}early-boot-config
+Requires: %{_cross_os}early-boot-config-common
+Requires: %{_cross_os}early-boot-config-data-providers
+Requires: %{_cross_os}early-boot-config-local
+%description -n %{_cross_os}early-boot-config-vmware
+%{summary}.
+%endif
+
+%package -n %{_cross_os}early-boot-config-metal
+Summary: early-boot-config package for metal
+Provides: %{_cross_os}early-boot-config
+Requires: %{_cross_os}early-boot-config-common
+Requires: %{_cross_os}early-boot-config-data-providers
+Requires: %{_cross_os}early-boot-config-local
+%description -n %{_cross_os}early-boot-config-metal
+%{summary}.
+
+%prep
+%setup -T -c
+%cargo_prep
+
+%build
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p early-boot-config \
+    --bin early-boot-config
+
+# build aws user data providers
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p ec2-identity-doc-user-data-provider \
+    -p ec2-imds-user-data-provider
+
+# build local file user data providers
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p local-defaults-user-data-provider \
+    -p local-file-user-data-provider \
+    -p local-overrides-user-data-provider
+
+%ifarch x86_64
+# build vmware user data providers
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p vmware-cd-rom-user-data-provider \
+    -p vmware-guestinfo-user-data-provider
+%endif
+
+%install
+install -d %{buildroot}%{_cross_bindir}
+install -p -m 0755 ${HOME}/.cache/%{__cargo_target}/release/early-boot-config %{buildroot}%{_cross_bindir}
+
+install -d %{buildroot}%{_cross_unitdir}
+install -p -m 0644 %{S:100} %{buildroot}%{_cross_unitdir}
+
+install -d %{buildroot}%{_cross_libexecdir}/early-boot-config/bin
+install -p -m 0755 \
+    ${HOME}/.cache/%{__cargo_target}/release/ec2-identity-doc-user-data-provider \
+    ${HOME}/.cache/%{__cargo_target}/release/ec2-imds-user-data-provider \
+    ${HOME}/.cache/%{__cargo_target}/release/local-defaults-user-data-provider \
+    ${HOME}/.cache/%{__cargo_target}/release/local-file-user-data-provider \
+    ${HOME}/.cache/%{__cargo_target}/release/local-overrides-user-data-provider \
+    %{buildroot}%{_cross_libexecdir}/early-boot-config/bin
+
+%ifarch x86_64
+install -p -m 0755 \
+    ${HOME}/.cache/%{__cargo_target}/release/vmware-cd-rom-user-data-provider \
+    ${HOME}/.cache/%{__cargo_target}/release/vmware-guestinfo-user-data-provider \
+    %{buildroot}%{_cross_libexecdir}/early-boot-config/bin
+%endif
+
+install -d %{buildroot}%{_cross_datadir}/early-boot-config/data-providers.d
+
+%post -n %{_cross_os}early-boot-config-aws -p <lua>
+posix.symlink("../../../libexec/early-boot-config/bin/ec2-identity-doc-user-data-provider", "%{_cross_datadir}/early-boot-config/data-providers.d/30-ec2-identity-doc")
+posix.symlink("../../../libexec/early-boot-config/bin/ec2-imds-user-data-provider", "%{_cross_datadir}/early-boot-config/data-providers.d/40-ec2-imds")
```

Comment:
There's no conflict between any of these symlinks from what I can see, so I would skip the `%post` scriptlets and just create and package the symlinks:
```
# before %install
# convention is not to include %{buildroot} - then you can reuse these macros in the %files sections.

%global early_boot_config_bindir %{_cross_libexecdir}/early-boot-config/bin
%global early_boot_config_provider_dir %{_cross_datadir}/early-boot-config/data-providers.d

# instead of %post
# (`ln -r` will take care of the relative symlink for you.)

install -d %{buildroot}%{early_boot_config_provider_dir}
ln -rs \
  %{buildroot}%{early_boot_config_bindir}/ec2-identity-doc-user-data-provider 
  %{buildroot}%{early_boot_config_provider_dir}/40-ec2-imds
```